src/NuGet.Licenses/Controllers/LicenseController.cs

```diff
@@ -151,6 +151,25 @@ private ActionResult DisplayCompositeLicenseExpression(LicenseOperator licenseEx
             var meaningfulSegments = _licenseExpressionSegmentator.GetLicenseExpressionSegments(licenseExpressionRoot);
             var allSegments = _licenseExpressionSegmentator.SplitFullExpression(licenseExpression, meaningfulSegments);
 
+            foreach(var segment in allSegments)
+            {
+                if (segment.Type == CompositeLicenseExpressionSegmentType.LicenseIdentifier || segment.Type == CompositeLicenseExpressionSegmentType.ExceptionIdentifier)
+                {
+                    try
+                    {
+                        if (!_licenseFileService.DoesLicenseFileExist(segment.Value))
+                        {
```

Comment:
I am a little bit concerned about doing a disk operation for each license or exception in the list. As discussed in person, this can be improved by checking what parser thinks of it.
I see several ways to do it:
* Just feed the `segment.Value` into the `NuGetLicenseExpression.Parse` and check the output (so it has `IsStandardLicense`, `IsFsf` and `IsOsi` specified before checking for file existence);
* Traverse the expression tree (similar to how [Segmentator](https://github.com/NuGet/NuGet.Licenses/blob/b460c8cc178a406f756a8f288687fa9a8e574745/src/NuGet.Licenses/Services/LicenseExpressionSegmentator.cs#L80) does it);
* Change the definition of `CompositeLicenseExpressionSegment` to include some additional information and change the segmentator code to put that information there.

Alternatively, leave everything as is, but add some telemetry around how much time does it take and how much time `DoesLicenseFileExist` is called per request and we can monitor if an optimization is required here. Test the worst case (`MIT OR MIT OR MIT...` up to expression length limit) and check if there is a noticeable perf impact...

Reply:
Pick up the simplest 1 choice, and add the test cases. 
There are still some considerations here:
1. We will for-loop the segments to valid each one. The current logic is that once we find one unavailable license or exception, we will return "404 unknownLicense". But for the case "MIT AND unknownlicense1 AND unknownlicense2", we will only return the first unknownlicense name when it's reached, is this expected by customers? I prefer to return 400 general error "InvalidRequest()" without providing the explicit license name.
2. Do we really need to check whether license file exists for composite license expression function. We only parse the composite license expression and provide possible links in that function. I think in this place we do not need to check if they exist, and we only check whether the license is a valid one or not. Then, if the license file does not exist, when the customer click the link and try to fetch the license content, we will return "404 unknownLicense" with the message. I think this is clear for the customer that the name is good but we don't have the information and we can avoid the heavy load of the I/O operations for checking license path.